### PR TITLE
Unit tests for values in indicator records.

### PR DESCRIPTION
**Pull request by sheldonkreger**

### CODE REVIEW

File: tests/test_pivoteer.py [hunk 1]
```diff
@@ -0,0 +1,195 @@
...
+from django.test import TestCase
+from pivoteer.models import IndicatorRecord, IndicatorManager
+from pivoteer.tasks import certificate_cen, domain_thc, ip_thc, domain_whois, ip_whois, domain_hosts, ip_hosts, passive_hosts, malware_samples, google_safebrowsing, totalhash_ip_domain_search, make_indicator_search_records
+
+app = Celery('RAPID')
+
+logger = logging.getLogger(None)
+
+
+class SimplisticTest(TestCase):
```
Comment (sbgoodm) [inline tests/test_pivoteer.py]: Would you rename this class to something more descriptive of its purpose?

**Comment by bauerm** [comment]
These appear to be hitting live APIs, which is both going to count against our api limits and be inconsistent (data may change, or services may not be available)

We need to mock the webservices and use fixed results.

**Comment by sheldonkreger** [comment]
Given that the end users will be hitting the APIs hundreds or perhaps thousands of times per day, I don't see any disadvantage to hitting them while running tests. Plus, if an API changes and breaks a tests, that's worth catching. Technically that's more of a functional test than a unit test, but I see value in it regardless.

What's the advantage to mocking the web services?

**Comment by bauerm** [comment]
You can test the data against the expected result if you hit a mock service, hitting a live service whose results may/will change doesn't allow you to test for appropriate results. You need a fixed input -> code under test -> expected output ... without that fixed input, you can't test an expected output.

**Comment by sheldonkreger** [comment]
I see the issue of unpredictable results from the API. I've confused the roles of integration and unit tests here.

What is the best way to mock up the results? I could write directly to the DB during the setup() method and load the appropriate results in each test. Or, I could implement another class that just returns the data.

The problem with these approaches is that I'm only testing the 'load' portion . . . not the 'save' portion. Not sure if it's feasible to mock up exact HTTP responses from each API so we can test the 'save' portion accurately.

**Comment by sbgoodm** [comment]
I'm not sure on the 'how' of doing this in Python/Django, but I have mocked up http responses from an external service in Java unit tests many times. This is done by creating a mock object that always returns a fixed value, and inserting this mock object in the place of the real object. 

This page has some good details -  https://docs.python.org/3/library/unittest.mock.html#unittest.mock.Mock

Particularly the section on Autospeccing

**Comment by bauerm** [comment]
+1

**Pull request merged**